sql/pgwire/encoding.go

```diff
@@ -78,7 +59,11 @@ func (b *readBuffer) readUntypedMsg(rd io.Reader) error {
 			size, maxMessageSize)
 	}
 
-	b.reset(size)
+	if len(b.msgBuf) < size {
```

Comment:
It's once per incoming *message*, not request (one logical request may contain multiple messages), but I still think it's safe to reuse `msgBuf`. Anything that would make it unsafe would be a bad idea even before this change (holding a reference to an unnecessarily-large chunk of memory). OTOH, such problems can be difficult to detect and so they may exist; it may not be worth making this change if it doesn't even show up in the `SELECT 1` benchmark. 

I'm concerned about `msgBuf` getting too big: with this change, each connection will contain a buffer that grows to the size of the largest message it has received. I'd rather make the reusable buffers fixed-size (4096 bytes?) and just allocate a one-off buffer when we need something larger. (would it make more sense to put them in a sync.Pool instead of the connection itself?)